lib/tasks/erc_auth.rake

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require "csv"
+require "file_manager"
+
+namespace :erc_auth do
+  include FileManager
+
+  namespace :csv_import do
+    task all: [:comarcals, :regionals, :locals, :local_comarcal_relationships, :local_regional_relationships]
+
+    desc "Generates a YAML file with the CSV with 'Comarcal' tag"
+    task comarcals: :environment do
+      csv_text = File.read("tmp/scopes_codes.csv")
+      csv = CSV.parse(csv_text, headers: true)
+
+      csv = csv.select { |row| row["type"] == "FC" }
+      csv = csv.map(&:to_h)
+
+      result = get_code_and_display_name(csv)
+      write_config_yaml!("comarcals", result)
+    end
+
+    desc "Generates a YAML file with the CSV with 'Regional' tag"
+    task regionals: :environment do
+      csv_text = File.read("tmp/scopes_codes.csv")
+      csv = CSV.parse(csv_text, headers: true)
+
+      csv = csv.select { |row| row["type"] == "FR" }
+      csv = csv.map(&:to_h)
+
+      result = get_code_and_display_name(csv)
+      write_config_yaml!("regionals", result)
+    end
+
+    desc "Generates a YAML file with the CSV with 'Locals' tag"
+    task locals: :environment do
+      csv_text = File.read("tmp/scopes_codes.csv")
+      csv = CSV.parse(csv_text, headers: true)
+
+      csv = csv.select { |row| row["type"] == "SL" }
+      csv = csv.map(&:to_h)
+
+      result = get_code_and_display_name(csv)
+      write_config_yaml!("locals", result)
+    end
+
+    desc "Generates a YAML file with the relationship between CSV 'Local' and 'Comarcal'"
+    task local_comarcal_relationships: :environment do
```

Comment:
isn't this task file the same as in `civi_crm`?

Reply:
I not the same, look:

in cvm_crm:

**data = get_civi_crm_data(:find_local_organization_relationships)**
comarcal_ids = load_config_yaml("comarcals").keys
**result = get_contact_id_relationships(data, comarcal_ids)**
write_config_yaml!("local_comarcal_relationships", result)

in erc_auht:

**local_ids = load_config_yaml("locals").keys**
comarcal_ids = load_config_yaml("comarcals").keys
**result = get_code_relationships(local_ids, comarcal_ids, "comarcal")**
write_config_yaml!("local_comarcal_relationships", result)

The same happens in local_regional_relationships